File: vb/SolidEdge/ExtractBOM/ExtractBOM/BomItem.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace ExtractBOM
    Public Class BomItem
        Private _children As New List(Of BomItem)()

        Public Sub New()
        End Sub

        Public Sub New(ByVal occurrence As SolidEdgeAssembly.Occurrence)
            FileName = occurrence.OccurrenceFileName
        End Sub

        Public FileName As String
        Public Missing? As Boolean
        Public Property Occurrence() As List(Of BomItem)
            Get
                Return _children
            End Get
            Set(ByVal value As List(Of BomItem))
                _children = value
            End Set
        End Property

        Public Overrides Function ToString() As String
            Return FileName
        End Function
    End Class
End Namespace

File: cs/SolidEdge/ExtractBOM/ExtractBOM/BomItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExtractBOM
{
    public class BomItem
    {
        private List<BomItem> _children = new List<BomItem>();

        public BomItem()
        {
        }

        public BomItem(SolidEdgeAssembly.Occurrence occurrence)
        {
            FileName = occurrence.OccurrenceFileName;
        }

        public string FileName;
        public bool? Missing;
        public List<BomItem> Occurrence { get { return _children; } set { _children = value; } }

        public override string ToString()
        {
            return FileName;
        }
    }
}
